hubspot3/test/test_tickets.py

```diff
@@ -48,15 +48,72 @@ def test_get_ticket():
     assert isinstance(ticket, dict)
 
 
-def test_get_all_tickets():
+def test_get_all_tickets(tickets_client, mock_connection):
     """
     tests getting all tickets
     :see: https://developers.hubspot.com/docs/methods/tickets/get-all-tickets
     """
-    tickets = TICKETS.get_all()
-    assert tickets
-    assert isinstance(tickets, list)
-    assert len(tickets) > 1
+    response_bodies = [{
+        "objects": [
+            {
+                "addedAt": 1390574181854,
+                "objectId": 204726,
+                "properties": {'prop_1': 1},
+            }
+        ],
+        "hasMore": True,
+        "offset": 204727,
+    }, {'objects': [], "hasMore": True, "offset": 204727}, {
+        "objects": [
+            {
+                "addedAt": 1390574181854,
+                "objectId": 204727,
+                "properties": {'prop_1': 1},
+            }
+        ],
+        "hasMore": False,
+        "offset": 204727,
+    }, {
+        "objects": [
+            {
+                "addedAt": 1390574181854,
+                "objectId": 204727,
+                "properties": {'prop_2': 1},
+            },
+            {
+                "addedAt": 1390574181854,
+                "objectId": 204728,
+                "properties": {'prop_2': 1},
+            }
+        ],
+        "hasMore": False,
+        "offset": 204727,
+    }]
+
+    # This extra properties are enough to generate 2 requests
+    extra_properties = [str(num) for num in range(2000)]
+
+    responses = [(200, response_body) for response_body in response_bodies]
+    mock_connection.set_responses(responses)
+
+    tickets = tickets_client.get_all(extra_properties=extra_properties)
+
+    mock_connection.assert_num_requests(4)
+    for extra_property in extra_properties:
+        # This checks if a request had the URL and includes the given parameters
+        mock_connection.assert_has_request(
+            "GET", "/crm-objects/v1/objects/tickets/paged", **{'properties': extra_property}
+        )
+
+    assert len(tickets) == 3
+
+    first_ticket = [ticket for ticket in tickets if ticket['objectId'] == 204726][0]
+    second_ticket = [ticket for ticket in tickets if ticket['objectId'] == 204727][0]
+    third_ticket = [ticket for ticket in tickets if ticket['objectId'] == 204728][0]
+
+    assert list(first_ticket['properties'].keys()) == ['prop_1']
+    assert list(second_ticket['properties'].keys()) == ['prop_1', 'prop_2']
```

Comment:
You cannot guarantee the order of keys in a dict. You should test against sets (which need no conversion):

`assert second_ticket['properties'].keys() == {'prop_1', 'prop_2'}`